runtime/assembly/modules/storage.ts

```diff
@@ -36,4 +38,48 @@ export namespace Storage {
         const value64: u64 = Serialiser.serialiseResultwOutRetain(value);
         ext_storage_set_version_1(key64, value64);
     }
+    /**
+     * Gets the given key from storage, placing the value into a buffer and returning the number of bytes that the
+     * entry in storage has beyond the offset. 
+     * ArrayBuffer has fixed length of 20.
+     * Returns Option instance with populated ArrayBuffer concatenated to the result from storage_read
+     * @param key key of the data to get
+     * @param offset offset beyond which the value is copied
+     */
+    export function read(key: u8[], offset: i32): Option<u8[]> {
```

Comment:
I think that when we use the `Storage.read` function we will supply the `array buffer` from outside this function. This is a low-level function in which you cannot determine what the size of the buffer will be. You don't know whether you are reading a large data or a 3-byte memory. That;s why I think that the buffer should be passed as argument from the outside, otherwise this read function will not be practical.

In that case, I think that it will be better to return the `result` from the read, meaning that we should return the number of bytes that are stored at that key or return `null` (`None`) if the entry does not exist.